packages/framework/ai-collab/src/test/explicit-strategy/agentEditing.spec.ts

```diff
@@ -0,0 +1,271 @@
+
+const demoSf = new SchemaFactory("agentSchema");
+
+class Vector extends demoSf.object("Vector", {
+	x: demoSf.number,
+	y: demoSf.number,
+	z: demoSf.optional(demoSf.number),
+}) {}
+
+class RootObject extends demoSf.object("RootObject", {
+	str: demoSf.string,
+	vectors: demoSf.array(Vector),
+	bools: demoSf.array(demoSf.boolean),
+}) {}
+
+const factory = SharedTree.getFactory();
+
+describe("toDecoratedJson", () => {
+	let idGenerator: IdGenerator;
+	beforeEach(() => {
+		idGenerator = new IdGenerator();
+	});
+	it("adds ID fields", () => {
+		const tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: Vector }));
+		view.initialize({ x: 1, y: 2 });
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({
+				[objectIdKey]: "Vector1",
+				x: 1,
+				y: 2,
+			}),
+		);
+	});
+
+	it("handles nested objects", () => {
+		const tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: RootObject }));
+		view.initialize({ str: "hello", vectors: [{ x: 1, y: 2, z: 3 }], bools: [true] });
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({
+				[objectIdKey]: "RootObject1",
+				str: "hello",
+				vectors: [
+					{
+						[objectIdKey]: "Vector1",
+						x: 1,
+						y: 2,
+						z: 3,
+					},
+				],
+				bools: [true],
+			}),
+		);
+
+		assert.equal(idGenerator.getNode("RootObject1"), view.root);
+		assert.equal(idGenerator.getNode("Vector1"), view.root.vectors.at(0));
+
+		// const hydratedObject = hydrate(
+		// 	RootObject,
+		// 	new RootObject({ str: "hello", vectors: [{ x: 1, y: 2, z: 3 }], bools: [true] }),
+		// );
+		// assert.equal(
+		// 	toDecoratedJson(idGenerator, hydratedObject),
+		// 	JSON.stringify({
+		// 		[objectIdKey]: "RootObject0",
+		// 		str: "hello",
+		// 		vectors: [
+		// 			{
+		// 				[objectIdKey]: "Vector0",
+		// 				x: 1,
+		// 				y: 2,
+		// 				z: 3,
+		// 			},
+		// 		],
+		// 		bools: [true],
+		// 	}),
+		// );
+		// assert.equal(idGenerator.getNode("RootObject0"), hydratedObject);
+		// assert.equal(idGenerator.getNode("Vector0"), hydratedObject.vectors.at(0));
+	});
+
+	it("handles non-POJO mode arrays", () => {
+		const sf = new SchemaFactory("testSchema");
+		class NamedArray extends sf.array("Vector", sf.number) {}
+		class Root extends sf.object("Root", {
+			arr: NamedArray,
+		}) {}
+
+		const tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		const view = tree.viewWith(new TreeViewConfiguration({ schema: Root }));
+		view.initialize({ arr: [1, 2, 3] });
+
+		assert.equal(
+			toDecoratedJson(idGenerator, view.root),
+			JSON.stringify({ __fluid_objectId: "Root1", arr: [1, 2, 3] }),
```

Comment:
The tests above use this, so this one should too, right?
```suggestion
			JSON.stringify({ [objectIdKey]: "Root1", arr: [1, 2, 3] }),
```

Reply:
Updated in commit 99bdf91a942f62d35c650e62bb0b5debc8dfa00c